modules/core/src/test/java/org/apache/ignite/internal/GridDiscoveryManagerChangeCoordinatorTest.java

```diff
@@ -0,0 +1,197 @@
+     */
+    @Test
+    public void testChangeCoordinatorLogging() throws Exception {
+        //Start 2 server nodes
+        IgniteEx srv1 = startGrid("server1");
+        startGrid("server2");
+
+        srv1.cluster().active();
+
+        IgniteEx client = startGrid("client");
+
+        stopGrid("server1");
+        srv1 = startGrid("server1");
+
+        stopGrid("server2");
+        startGrid("server2");
+
+        UUID clientClusterNode = client.localNode().id();
+        UUID srv1ClusterNode = srv1.localNode().id();
+
+        Pattern ptrn = Pattern.compile(String.format(CRD_CHANGE_MSG, srv1ClusterNode, clientClusterNode));
+
+        LogListener lsnr = LogListener.matches(ptrn).build();
+
+        listeningLog.registerListener(lsnr);
+
+        stopGrid("client");
+
+        // Check that there are no messages like "coordinator changed server1 -> client"
+        assertFalse(lsnr.check());
+    }
+
+    /**
+     * Checks change coordinator event logging with daemon node in topology.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testChangeCoordinatorLoggingDaemonNodeInTopology() throws Exception {
+        // Start 2 server nodes
+        IgniteEx srv1 = startGrid("server1");
+        IgniteEx srv2 = startGrid("server2");
+
+        srv1.cluster().active();
+
+        // Check coordinator = server1
+        assertTrue(((TcpDiscoverySpi) srv1.context().config().getDiscoverySpi()).isLocalNodeCoordinator());
+
+        // Add a client node, daemon node and 3-d server node, which is not in the baseline
+        startGrid("client");
+        IgniteEx daemon = startGrid("daemon");
+        IgniteEx srv3 = startGrid("server3");
+
+        UUID srv1ClusterNode = srv1.localNode().id();
+        UUID srv2ClusterNode = srv2.localNode().id();
+
+        Pattern ptrn = Pattern.compile(String.format(CRD_CHANGE_MSG, srv1ClusterNode, srv2ClusterNode));
+
+        LogListener lsnr = LogListener.matches(ptrn).build();
+
+        listeningLog.registerListener(lsnr);
+
+        stopGrid("server1");
+
+        UUID crdUUID = ((TcpDiscoverySpi)srv3.context().config().getDiscoverySpi()).getCoordinator();
+
+        // Coordinator changed server1 -> server2
+        assertEquals(srv2.localNode().id(), crdUUID);
+
+        // Check that there is message like "coordinator changed server1 -> server2"
+        assertTrue(lsnr.check());
+
+        ptrn = Pattern.compile(String.format(CRD_CHANGE_MSG, srv2ClusterNode, srv1ClusterNode));
+
+        lsnr = LogListener.matches(ptrn).build();
+
+        listeningLog.registerListener(lsnr);
+
+        srv1 = startGrid("server1");
+
+        crdUUID = ((TcpDiscoverySpi) srv3.context().config().getDiscoverySpi()).getCoordinator();
+
+        // Server2 is still the coordinator
+        assertEquals(srv2.localNode().id(), crdUUID);
+
+        // Check that there are no messages like "coordinator changed server2 -> server1"
+        assertFalse(lsnr.check());
```

Comment:
same as upper: no one occurrence of "Coordinator changed" need to be detected here.